docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,434 @@
+
+Several mechanisms can be used to track external references:
+* Polling
+* Webhooks
+
+#### `/device/{name}/rendered` (device side)
+This endpoint is used by the device agent to retrieve the rendered device spec, this endpoint is
+read-only, each device can only access it's own device rendered spec based on the device
+certificate identification, an administrator can access any device rendered spec for debug purposes.
+
+The agent can pass two parameters: `version`and `owner` if those haven't changed the API will return
+204 No Content, if the version or fleet has changed the API will return the rendered spec.
+
+Change proposed:
+* The parameter name `fleet` is changed to `owner` to be consistent with the metadata field.
+
+Behavior changes:
+* If a device configuration is not yet rendered for some reason, FC will queue the device for rendering and instruct
+  the device to try again later by returning 202 Accepted.
+
+The administrator can find the fully rendered device spec by using the following cmdline command:
+```shell
+flightctl get device/xxxxxxx --rendered
+```
+
+The JSON blob configuration exposed to the device in the /device/{name}/rendered endpoint
+while compacted for transport, should be formated in a way that it's easy to read by humans
+when printed out to the console, by adding tabs and newlines.
+
+#### `/device/{name}/status` (device side)
+
+This endpoint is used by the device agent to report the status of the device, this endpoint is write-only,
+each device can only access it's own device status based on the device certificate identification.
+
+The result of this write will be exposed on the `/device/{name}` endpoint as part of the status field.
+
+Reporting to this endpoint also serves as a heartbeat mechanism, the device agent should report the status,
+if a device stops reporting to this endpoint in the expected timelines will be considered offline.
+
+NOTE: When optimization is required in the future we could implement an additional endpoint to report heartbeats,
+i.e. `/device/{name}/heartbeat` to reduce the amount of data being sent to the server. This heartbeat could include
+a hash or fingerprint of the device status content. The server could request the device to post full status (i.e.
+if the status on the server went stale, was lost or restored from backup).
+
+### Fleet API
+
+Fleet is an automation layer on top of Device API, the DeviceTemplate is a mechanism to control drift
+and enforce policy across devices in a fleet.
+
+A DeviceTemplate is a template for a device spec, it's a way to define group specific configuration.
+
+We are proposing extension mechanisms to this API to allow for more complex configurations where
+device labels or name would influence the configuration generated to the device.
+
+https://github.com/flightctl/flightctl/pull/248
+
+#### Expected behavior
+
+When fleet.spec.template.metadata or fleet.spec.template.spec is updated, the Flet deployment controller
```

Comment:
```suggestion
When fleet.spec.template.metadata or fleet.spec.template.spec is updated, the Fleet deployment controller
```

Reply:
Shorten to `fleet.spec.template`? Or do we expect anything under that which does not cause an update?